securityinternal/v1/types.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RangeAllocation is used so we can easily expose a RangeAllocation typed for security group
+// This is an internal API, not intended for external consumption.
+type RangeAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// range is a string representing a unique label for a range of uids, "1000000000-2000000000/10000".
+	Range string `json:"range"`
+
+	// data is a byte array representing the serialized state of a range allocation.  It is a bitmap
+	// with each bit set to one to represent a range is taken.
+	Data []byte `json:"data"`
```

Comment:
this is really an array of decimal bytes? Why no base64 string?

Not super important, just super ugly.

Reply:
> this is really an array of decimal bytes? Why no base64 string?
> 
> Not super important, just super ugly.

This won't be changed.  I seem to recall it truly being a bitmap.

Reply:
Don't change that, it has to be identical with the original.